### ISSUE TITLE
Kernel Panic after updating from unstable branch

### ISSUE THREAD

**Issue by qnfbIbgUDItR14NR** [rockstor/rockstor-core#1085]
Screenshot of kernel panic:  http://i.imgur.com/yh4DQ0o.jpg]

This occurs immediately after GRUB goes away.

Hardware
* CPU:  Athlon 5350
* Motherboard:  ASRock AM1H-ITX
(I should note that two of the SATA3 ports are provided natively by a SATA controller on the CPU.  The other two are provided by an ASMedia controller on the motherboard.  I haven't noticed any incompatibilities due to this.)
* RAM:  Klevv Fit 8GB (2 x 4GB) KM3F4GX2C-1600-09-09-09-24-0
* Boot drive:  Hitachi 5400 RPM 160 GB 2.5" drive (don't have precise model number handy)
* Other drives:  Two (2x) Seagate ST4000VX000 4TB

I installed Rockstor 3.8-10 from a flash drive and had a functional home NAS.  I updated once sometime yesterday from the unstable branch and then once again today (after performing a fresh reinstall), both times getting the same kernel panic.  I've tried disabling unnecessary features in the BIOS such as onboard audio to no avail (besides, RockStor was working before the update.)

Kernels available to me are (and these are verbatim from GRUB)
* Rockstor (4.3.3-1.el7.elrepo.x86_64) 3 (Core) with debugging
* Rockstor (4.3.3-1.el7.elrepo.x86_64) 3 (Core)
* Rockstor 3 (Core), with Linux 4.2.5-1.el7.elrepo.x86_64
* Rockstor 3 (Core), with Linux 0-rescue-a353ff50d60a4c9da12cac803dc9ce3a

The first two options available to me instantly result in a kernel panic.  The third one functions.  I have not tried the 4th.

If there's any other information that I can provide, I'd be more than happy to.

**Comment by qnfbIbgUDItR14NR** [comment]
I can manually select the previous kernel and successfully boot, however, the web UI seems to be inaccessible now.  That is, browsers will time out when pointed to the appropriate address.

This is interesting because the box still responds to pings and I can SSH onto the box.

**Comment by phillxnet** [comment]
@qnfbIbgUDItR14NR I'm afraid there is little we can do re the kernel issue as we simply use the elrepo kernel-ml but given for the time being the 4.2.5 option is working for you bar the UI I'm guessing you have two issues here. During a recent upgrade the python Rockstor depends upon was upgraded and some Rockstor incompatible backports were introduced. This may be the reason for your UI failing. This same update cycle introduced the newer kernel.

The following command is worth a try as in some cases the work around introduced in Rockstor to do the same seems not to have taken effect in time:-
```
yum downgrade python-2.7.5-18.el7_1.1.x86_64 python-libs-2.7.5-18.el7_1.1.x86_64
```
Hopefully that will sort your No UI issue.
There is a forum post that I suspect is the same situation re no UI after what I think may be the same upgrade:-
http://forum.rockstor.com/t/shares-despairing-and-not-no-web-control/817/2

As for the later kernels we do have other reports of this happening and funnily enough an almost identical kernel panic screen in this forum post:- 
http://forum.rockstor.com/t/kernel-panic-after-upgrade-to-3-8-10/818
Except in @rayburgemeestre forum members case it was with the 4.2.5 kernel. Kernel compatibility / hw issues are really outside the scope of Rockstor but given the elrepo kernel is used there may well be benefit in searching for the specific elrepo kernel and hardware for others having the same problem.

As for temporarily altering Rockstor to auto boot to the working kernel until a newer and hopefully also working kernel arrives in Rockstor please note the second comment by suman of @schakrava in the latter above forum thread for some more context on this. Also see the following forum thread:-
http://forum.rockstor.com/t/override-kernel-autoboot/798 to sort the auto selection of the kernel that works for you.
Obviously this is not ideal but may get you up and running for the time being.

Thanks for offering to help and it's worth noting that Rockstor puts all it's logs in:-
/opt/rockstor/var/log/
and the main file to watch re more info is:-
/opt/rockstor/var/log/rockstor.log

Hope this helps and let us know how this works out for you.

**Comment by qnfbIbgUDItR14NR** [comment]
Thanks @phillxnet for the information.

I have, in the interim, done a fresh reinstall of Rockstor and will be holding on that version for a bit.  Probably in a few months I will take the plunge and update once again to see if the kernel available at that time resolves my issues.

If I once again run into issues at that time, I will share what information I have and keep the box in a nonfunctional state longer in case there are any logfiles or dumps that would be of use to you and the team.

That being said, I understand that kernel issues are a bit upstream and out of your control, so if it's appropriate to close out this issue, I fully understand.  Thanks for your help!  I settled on Rockstor due to Linux having support for a wider range of hardware (over the BSD family), and while I have no personal experience with prior versions of Rockstor, basic research tells me this project is rapidly maturing, and I can't wait to see what you all cook up next.

I might also add that the reinstallation process was painless, and picking up right where I left off was fairly easy (though the UI didn't make it totally obvious that I could just import the pool... I only noticed it because I was in "mouse over all the icons" mode.  Still, it "just worked"!)

**Comment by qnfbIbgUDItR14NR** [comment]
Just wanted to close this issue out by saying I have successfully updated to the latest unstable version of Rockstor (3.8-10.19).  Yes, I am aware 3.8-11 stable was released today =p

Before updating from the web UI, I did ssh onto the box and run a `yum update`.  After that completed, I rebooted from the web UI and the system came back up successfully, though it did complain about an unsupported kernel.  Given that last time, I couldn't even boot with the `4.3.3-1.el7.elrepo.x86_64` kernel, I decided to update from inside the web UI.  That completed successfully.
